### PR TITLE
cell_locations: Writing output to S3 was failing; fixed this

### PR DESCRIPTION
**Pull request by shntnu**
_modified from [EmbeddedArtistry](https://embeddedartistry.com/blog/2017/08/04/a-github-pull-request-template-for-your-projects/)_

# Description

The code failed when the output was on S3; I needed to fix a bug that checks for the file's existence on S3.

## What is the nature of your change?

- [x] Bug fix (fixes an issue).

# Checklist

Please ensure that all boxes are checked before indicating that a pull request is ready for review.

- [x] I have read the [CONTRIBUTING.md](CONTRIBUTING.md) guidelines.
- [x] My code follows the style guidelines of this project.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [x] New and existing unit tests pass locally with my changes.
- [x] I have added tests that prove my fix is effective or that my feature works. No; See #265
- [x] I have deleted all non-relevant text in this pull request template.

### CODE REVIEW

**Comment by shntnu** [comment]
@d33bs please note that I could not add a test to check for this; see #265

File: pycytominer/cyto_utils/cell_locations.py [hunk 1]
```diff
@@ -147,7 +147,7 @@ def _s3_file_exists(self, s3_path: str):
         bucket, key = self._parse_s3_path(s3_path)
 
         try:
-            self.s3.Object(bucket, key).load()
+            self.s3.head_object(Bucket=bucket, Key=key)
         except botocore.exceptions.ClientError as e:
```
Comment (d33bs) [inline pycytominer/cyto_utils/cell_locations.py:151]: Double checking: does this exception get triggered by `boto3.s3.head_object()`?

File: pycytominer/cyto_utils/cell_locations.py [hunk 2]
```diff
@@ -147,7 +147,7 @@ def _s3_file_exists(self, s3_path: str):
...
             if e.response["Error"]["Code"] == "404":
```
Comment (d33bs) [inline pycytominer/cyto_utils/cell_locations.py:152]: While not a change for this PR, would we also want to catch `400`, or `403` errors somehow (recognizing they may result in similar circumstances)?

**Review by d33bs — CHANGES_REQUESTED** [review]
Thank you for this contribution @shntnu ! I'd like to request changes on this PR mostly with concerns to exception handling and general discussion.

Some additional items besides those mentioned in the comments:
- I'd ask that you please update the PR description with a root cause or need for the PR. It looks like the description still include the template content. If appropriate, please also update the PR title to the focus area as well. Doing this will help record what changes are occurring and why.
- Given that this occurred partially because of a lack of unit/functional tests, would it make sense to write a quick test to make sure we cover this scenario? Even a small test might provide enough coverage to incrementally begin addressing any unforeseen gaps.

What do you think? Interested in your thoughts - thank you in advance!

File: pycytominer/cyto_utils/cell_locations.py (hunk 2 above)
Comment (shntnu) [reply pycytominer/cyto_utils/cell_locations.py:152]: That makes sense; addressed

File: pycytominer/cyto_utils/cell_locations.py (hunk 1 above)
Comment (shntnu) [reply pycytominer/cyto_utils/cell_locations.py:151]: yes

**Comment by shntnu** [comment]
> * I'd ask that you please update the PR description with a root cause or need for the PR. It looks like the description still include the template content. If appropriate, please also update the PR title to the focus area as well. Doing this will help record what changes are occurring and why.

Done


> * Given that this occurred partially because of a lack of unit/functional tests, would it make sense to write a quick test to make sure we cover this scenario? Even a small test might provide enough coverage to incrementally begin addressing any unforeseen gaps.

Unfortunately, I don't have the expertise to write such a test because it involves the use of moto (#265)

I did create a stub for this here (see Details) https://github.com/cytomining/pycytominer/pull/257#issuecomment-1446184844 but I didn't have the expertise to push this further.

**Review by d33bs — APPROVED** [review]
Thanks @shntnu and apologies for my delay, this LGTM!

I do think we should prioritize testing to make sure work like this is covered by some form of testing in the future by addressing the issue you created. Just the same, I understand the need to move forward to balance progress + testing assurance.

@gwaybio - do you have any thoughts here as well?

**Review by gwaybio — APPROVED** [review]
LGTM! @shntnu - ok to merge?

**Comment by gwaybio** [comment]
actually, @shntnu - feel free to merge when you're happy (I assume you have write access to this repo).

**Comment by gwaybio** [comment]
@shntnu - doing some pycytominer cleanup. Is this good to merge?

**Comment by shntnu** [comment]
Oops, sorry. All set!

**Pull request merged**